src/rest/StreamEndpoints.js

```diff
@@ -1,45 +1,30 @@
 import qs from 'querystring'
-import { Agent as HttpAgent } from 'http'
-import { Agent as HttpsAgent } from 'https'
 import debugFactory from 'debug'
+import { authFetch } from './utils'
 
 import Stream from './domain/Stream'
-import { authFetch } from './utils'
 
 const debug = debugFactory('StreamrClient')
 
-const agentSettings = {
-    keepAlive: true,
-    keepAliveMsecs: 5000,
-}
-
-const agentByProtocol = {
-    http: new HttpAgent(agentSettings),
-    https: new HttpsAgent(agentSettings),
-}
-
-function getKeepAliveAgentForUrl(url) {
-    if (url.startsWith('https')) {
-        return agentByProtocol.https
-    } else if (url.startsWith('http')) {
-        return agentByProtocol.http
-    }
-
-    throw new Error(`Unknown protocol in URL: ${url}`)
-}
-
 // These function are mixed in to StreamrClient.prototype.
 // In the below functions, 'this' is intended to be the StreamrClient
 export async function getStream(streamId, apiKey = this.options.apiKey) {
     const url = `${this.options.restUrl}/streams/${streamId}`
-    const json = await authFetch(url, apiKey)
-    return json ? new Stream(this, json) : undefined
+    try {
```

Comment:
Why this special handling for 404? I think the `authFetch` (old and new version) throws if anything goes wrong (including 404), and that's an ok pattern to keep.

Maybe the answer to "Should an exception be thrown if a resource is not found?" is a matter of opinion, and certainly depends on whether the user of the function expects the resource to be found. For `getStream` he certainly does, as he passes an unguessable id to it => not finding the stream would certainly be an unexpected error situation.

(In the old code, I'm not sure under what conditions `json` could be falsey. Maybe it's a mistake.)

Reply:
I guess I got confused because of the checking `return json ? new Stream(this, json) : undefined`

Reply:
Also got confused because inside `getOrCreateStream` the call to `getStream` is not inside `try ... catch` (`client.getOrCreateStream({ id: 'foo', name: 'foo' })` will now throw an error) https://github.com/streamr-dev/streamr-client/pull/18/files#diff-759bd3eca5b1471aff3290ec566e0ed2L74